lib/spack/spack/environment.py

```diff
@@ -142,18 +141,17 @@ def activate(
             cmds += 'set prompt="%s ${prompt}";\n' % prompt
     else:
         if os.getenv('TERM') and 'color' in os.getenv('TERM') and prompt:
-            prompt = colorize('@G{%s} ' % prompt, color=True)
+            prompt = r"\[\033[0;92m\]{}\[\033[0m\]".format(prompt)
 
         cmds += 'export SPACK_ENV=%s;\n' % env.path
         cmds += "alias despacktivate='spack env deactivate';\n"
         if prompt:
-            cmds += 'if [ -z ${SPACK_OLD_PS1+x} ]; then\n'
-            cmds += '    if [ -z ${PS1+x} ]; then\n'
-            cmds += "        PS1='$$$$';\n"
-            cmds += '    fi;\n'
+            cmds += 'if [ -z "${SPACK_OLD_PS1}" ]; then\n'
             cmds += '    export SPACK_OLD_PS1="${PS1}";\n'
+            cmds += 'else\n'
+            cmds += '    export PS1="${SPACK_OLD_PS1}";\n'
             cmds += 'fi;\n'
-            cmds += 'export PS1="%s ${PS1}";\n' % prompt
+            cmds += 'export PS1="{} $PS1";\n'.format(prompt)
```

Comment:
This needs to be `{0}`, otherwise Python 2.6 won't work.

Reply:
Good catch. Thank you for pointing this out!